File: squares-of-a-sorted-array/squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
        int n = nums.size();
        int left = 0, right = n-1;
        vector<int> v(n, 0);
        for (int i = n-1; i>=0; i--) {
            if (abs(nums[left]) > abs(nums[right])) {
                v[i] = nums[left] * nums[left];
                left++;
            } else {
                v[i] = nums[right] * nums[right];
                right--;
            }
        }
        return v;
    }
};

File: squares-of-a-sorted-array/squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        import heapq
        squared_nums = list(map(lambda x: x**2, nums))
        heapq.heapify(squared_nums)
        return [heapq.heappop(squared_nums) for x in range(len(nums))]

